scripts/ImportExportTest.sh

```diff
@@ -0,0 +1,276 @@
+                cat obtained_direct_import_export.error >&2
+                FAILED=$((FAILED + 1))
+                return 0
+            fi
+
+            # reformat jsons using jq.
+            jq . expected.asm.json > expected.asm.json.pretty
+            jq . obtained_direct_import_export.json > obtained_direct_import_export.json.pretty
+            if ! diff_files expected.asm.json.pretty obtained_direct_import_export.json.pretty
+            then
+                _TESTED=
+                FAILED=$((FAILED + 1))
+                return 0
+            fi
+        fi
+    done
+
+    if [ -n "${_TESTED}" ]
+    then
+        TESTED=$((TESTED + 1))
+    fi
+}
+
+# function tests whether exporting and importing again leaves the JSON ast unchanged
+# Results are recorded by adding to FAILED or UNCOMPILABLE.
+# Also, in case of a mismatch a diff and the respective ASTs are printed
+# Expected parameters:
+# $1 name of the file to be exported and imported
+# $2 any files needed to do so that might be in parent directories
+function testImportExportEquivalence {
+    local sol_file="$1"
+    local input_files="$2"
+    if $SOLC --bin "${input_files}" > /dev/null 2>&1
+    then
+        ! [[ -e stderr.txt ]] || { fail "stderr.txt already exists. Refusing to overwrite."; }
+
+        if [[ $IMPORT_TEST_TYPE == "ast" ]]
+        then
+            Ast_ImportExportEquivalence "${sol_file}" "${input_files}"
+        elif [[ $IMPORT_TEST_TYPE == "evm-assembly" ]]
+        then
+            JsonEvmAsm_ImportExportEquivalence "${sol_file}" "${input_files}"
+        else
+            fail "Unknown import test type. Aborting."
+        fi
+    else
+        UNCOMPILABLE=$((UNCOMPILABLE + 1))
+    fi
+}
+
+WORKINGDIR=$PWD
+NSOURCES=0
+
+command_available "${SOLC}" --version
+command_available jq --version
+
+# for solfile in $(find $DEV_DIR -name *.sol)
+# boost_filesystem_bug specifically tests a local fix for a boost::filesystem
+# bug. Since the test involves a malformed path, there is no point in running
+# AST tests on it. See https://github.com/boostorg/filesystem/issues/176
+if [[ $IMPORT_TEST_TYPE == "ast" ]]
+then
+    TEST_DIRS=("${SYNTAXTESTS_DIR}" "${ASTJSONTESTS_DIR}")
+elif [[ $IMPORT_TEST_TYPE == "evm-assembly" ]]
+then
+    TEST_DIRS=("${SYNTAXTESTS_DIR}" "${SEMANTICTESTS_DIR}")
+else
+    fail "Unknown import test type. Aborting. Please specify ${0} [ast|evm-assembly]."
+fi
+
+IMPORT_TEST_FILES=$(find "${TEST_DIRS[@]}" -name "*.sol" -and -not -name "boost_filesystem_bug.sol")
+
+NSOURCES="$(echo "$IMPORT_TEST_FILES" | wc -l)"
+echo "Looking at $NSOURCES .sol files..."
+
+for solfile in ${IMPORT_TEST_FILES}
+do
+    echo -n "."
+    # create a temporary sub-directory
+    FILETMP=$(mktemp -d)
+    cd "$FILETMP"
+
+    set +e
+    OUTPUT=$("$SPLITSOURCES" "$solfile")
+    SPLITSOURCES_RC=$?
+    set -e
+    if [ ${SPLITSOURCES_RC} == 0 ]
+    then
+        OIFS=${IFS}
+        IFS=' ' read -ra OUTPUT_ARRAY <<< "${OUTPUT}"
+        IFS=${OIFS}
```

Comment:
You don't need to preserve the original value of `$IFS` here. The assignment takes effect only for the duration of the `read` command and then the original value is automatically restored.
```suggestion
        IFS=' ' read -ra OUTPUT_ARRAY <<< "${OUTPUT}"
```

Reply:
True!